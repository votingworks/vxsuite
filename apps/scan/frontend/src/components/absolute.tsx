import styled from 'styled-components';

interface Props {
  top?: boolean;
  right?: boolean;
  left?: boolean;
  bottom?: boolean;
  padded?: boolean;
}

export const Absolute = styled.div<Props>`
  position: absolute;
  inset: ${({ top }) => top && 0} ${({ right }) => right && 0} ${({ bottom }) => bottom && 0} ${({ left }) => left && 0};
  padding: ${({ padded }) => padded && '1rem'};
`;
