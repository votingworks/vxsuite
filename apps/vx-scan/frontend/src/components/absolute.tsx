import styled from 'styled-components';

interface Props {
  top?: boolean;
  right?: boolean;
  left?: boolean;
  bottom?: boolean;
  padded?: boolean;
}

export const Absolute = styled.div<Props>`
  position: absolute;
  top: ${({ top }) => top && 0};
  right: ${({ right }) => right && 0};
  bottom: ${({ bottom }) => bottom && 0};
  left: ${({ left }) => left && 0};
  padding: ${({ padded }) => padded && '1rem'};
`;
