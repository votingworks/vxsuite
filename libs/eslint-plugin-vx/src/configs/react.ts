import recommended = require('./recommended');

const jsExtensions = ['.js', '.jsx'];
const tsExtensions = ['.ts', '.tsx'];
const allExtensions = jsExtensions.concat(tsExtensions);

export = {
  ...recommended,
  extends: [
    ...recommended.extends
      .map((name) => (name === 'airbnb-base' ? 'airbnb' : name))
      .filter((name) => name !== 'plugin:prettier/recommended'),
    'airbnb/hooks',
    'plugin:react/recommended',
    'plugin:jsx-a11y/recommended',
    'plugin:prettier/recommended', // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
  ],
  settings: {
    ...recommended.settings,
    react: {
      version: 'detect',
    },
  },
  rules: {
    ...recommended.rules,
    'jsx-a11y/label-has-associated-control': [
      'error',
      {
        controlComponents: ['Select'],
      },
    ],
    'react/jsx-filename-extension': ['error', { extensions: allExtensions }],
    'react/jsx-fragments': ['error', 'element'],
    'react/jsx-no-bind': 'off',
    'react/jsx-no-constructed-context-values': 'off',
    'react/jsx-no-useless-fragment': ['error', { allowExpressions: true }],
    'react/jsx-one-expression-per-line': 'off',
    'react/jsx-props-no-spreading': 'off',
    'react/jsx-wrap-multilines': 'off',
    'react/prop-types': 'off',
    'react/require-default-props': 'off',
  },
};
