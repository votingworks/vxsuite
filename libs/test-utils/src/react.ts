/* eslint-disable no-console -- this file monkey-patches `console` */

/**
 * Suppresses the "Can't perform a React state update on an unmounted component" warning by
 * modifying the `console.error` method to ignore the warning.
 *
 * This is a temporary workaround until we upgrade to React 18.
 */
export function suppressReact17UnmountedWarning(): void {
  const originalConsoleError = console.error;
  console.error = (...args: unknown[]) => {
    if (
      typeof args[0] === 'string' &&
      args[0].startsWith(
        `Warning: Can't perform a React state update on an unmounted component`
      )
    ) {
      return;
    }

    originalConsoleError(...args);
  };
}
