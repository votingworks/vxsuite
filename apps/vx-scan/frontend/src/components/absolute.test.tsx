import React from 'react';
import { render } from '@testing-library/react';
import { Absolute } from './absolute';

test('Renders Absolute top right', () => {
  const { container } = render(<Absolute top right padded />);
  expect(container).toMatchSnapshot();
});

test('Renders Absolute bottom left', () => {
  const { container } = render(<Absolute bottom left />);
  expect(container).toMatchSnapshot();
});
