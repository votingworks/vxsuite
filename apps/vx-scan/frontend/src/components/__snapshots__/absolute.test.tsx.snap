// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Absolute bottom left 1`] = `
<div>
  <div
    class="sc-bdvvaa iWJoHm"
  />
</div>
`;

exports[`Renders Absolute top right 1`] = `
<div>
  <div
    class="sc-bdvvaa bdJOJp"
  />
</div>
`;
